source-code-flow-manager-gitlab/src/main/java/com/iprody/source/code/flow/manager/branch_api/BranchApi.java

```diff
@@ -0,0 +1,41 @@
+package com.iprody.source.code.flow.manager.branch_api;
+
+import com.iprody.source.code.flow.manager.branch_api.entity.Branch;
+import org.springframework.web.reactive.function.client.WebClient;
+import reactor.core.publisher.Mono;
+
+public class BranchApi {
+
+    private final String PRIVATE_TOKEN = "glpat-VwpTqGWV-SSQwV7xUypq";
+//    private final String BASE_URL = "https://gitlab.com/api/v4/projects/";
+
+    private WebClient webClient;
+
+    public BranchApi(String baseUrl) {
```

Comment:
This class accepts only Webclient that is preconfigured, as we discussed. No builder operation must be in these class, exception is a request preparation.

```suggestion
    public BranchApi(WebClient wc) {
```

Reply:
Yes, I agree!